frontend_server/package_incremental.py

```diff
@@ -0,0 +1,114 @@
+#!/usr/bin/env python
+# Copyright (c) 2012, the Dart project authors.  Please see the AUTHORS file
+# for details. All rights reserved. Use of this source code is governed by a
+# BSD-style license that can be found in the LICENSE file.
+
+import argparse
+import os
+import sys
+import shutil
+
+# The list of packages copied from the Dart SDK.
+PACKAGES = [
+    "vm", "build_integration", "kernel", "front_end", "frontend_server",
+]
+
+VM_PUBSPEC = r'''name: vm
+version: 0.0.1
+environment:
+  sdk: ">=2.2.2 <3.0.0"
+
+dependencies:
+  front_end: any
+  kernel: any
+  meta: any
+  build_integration: any
+'''
+
+BUILD_INTEGRATION_PUBSPEC = r'''name: build_integration
+version: 0.0.1
+environment:
+  sdk: ">=2.2.2 <3.0.0"
+
+dependencies:
+  front_end: any
+  meta: any
+'''
+
+FRONTEND_SERVER_PUBSPEC = r'''name: frontend_server
+version: 0.0.1
+environment:
+  sdk: ">=2.2.2 <3.0.0"
+
+dependencies:
+  args: any
+  path: any
+  vm: any
+'''
+
+KERNEL_PUBSPEC = r'''name: kernel
+version: 0.0.1
+environment:
+  sdk: '>=2.2.2 <3.0.0'
+
+dependencies:
+  args: any
+  meta: any
+'''
+
+FRONT_END_PUBSPEC = r'''name: front_end
+version: 0.0.1
+environment:
+  sdk: '>=2.2.2 <3.0.0'
+dependencies:
+  kernel: any
+  package_config: any
+  meta: any
+'''
+
+PUBSPECS = {
+    'vm': VM_PUBSPEC,
+    'build_integration': BUILD_INTEGRATION_PUBSPEC,
+    'frontend_server': FRONTEND_SERVER_PUBSPEC,
+    'kernel': KERNEL_PUBSPEC,
+    'front_end': FRONT_END_PUBSPEC,
+}
+
+def main():
+    parser = argparse.ArgumentParser()
+
+    parser.add_argument('--input-root', type=str, dest='input', action='store')
+    parser.add_argument('--output-root', type=str, dest='output', action='store')
+
+    args = parser.parse_args()
+    for package in PACKAGES:
+        package_root = os.path.join(args.input, package)
+        for root, directories, files in os.walk(package_root):
+            # We only care about actual source files, not generated code or tests.
+            for skip_dir in ['.git', 'gen', 'test']:
+                if skip_dir in directories:
+                    directories.remove(skip_dir)
+
+            # Ensure we have a dest directory
+            if not os.path.isdir(os.path.join(args.output, package)):
+                os.makedirs(os.path.join(args.output, package))
+
+            if '.git' in directories:
```

Comment:
Shouldn't this have been skipped already by the logic above?

Reply:
Fixed